docs_src/src/pages/documentation/api_reference/views.mdx

```diff
@@ -174,6 +174,143 @@ The only thing to remember is that you need to add the subrouter to the main rou
   </Col>
 </Row>
 
+### Authentication on Subrouters
+
+Batman asked Robyn how to secure his routes. Robyn explained that authentication can be added to subrouters in three ways, giving developers flexibility to decide the scope and precedence of the authentication. 
+
+> **Important**: If any of these methods are used, the authentication for the endpoint is set to `False` unless explicitly overridden.
+
+---
+<Row>
+  <Col>
+#### 1. Authentication at the Endpoint Level
+
+Authentication can be applied directly to a specific endpoint using the decorator for that endpoint. This method has the **highest precedence**, meaning it overrides any authentication settings defined at the router or subrouter level.
+</Col>
+<Col sticky>
+    <CodeGroup title="Request" tag="GET" label="/secured , /unsecured">
+```python {{ title: 'untyped' }}
+from robyn import Robyn, SubRouter
+
+app = Robyn(__file__)
+
+sub_router = SubRouter("/auth_example")
+
+@sub_router.get("/secured", auth_required=True)
+def secured(request):
+    return "This endpoint is secured with authentication!"
+
+@sub_router.get("/unsecured", auth_required=False)
+def unsecured(request):
+    return "This endpoint does not require authentication!"
+
+app.include_router(sub_router)
+```
+
+```python {{ title: 'typed' }}
+from robyn import Robyn, SubRouter, Request
+
+app = Robyn(__file__)
+
+sub_router = SubRouter("/auth_example")
+
+@sub_router.get("/secured", auth_required=True)
+def secured(request : Request) -> str:
+    return "This endpoint is secured with authentication!"
+
+@sub_router.get("/unsecured", auth_required=False)
+def unsecured(request : Request) -> str:
+    return "This endpoint does not require authentication!"
+
+app.include_router(sub_router)
+```
+    </CodeGroup>
+  </Col>
+</Row>
+
+
+<Row>
+  <Col>
+#### 2. Authentication via `include_router`
+When including a subrouter into the main router, you can specify an authentication flag. This setting applies to all endpoints in the subrouter unless overridden at the endpoint level.
+</Col>
+<Col sticky>
+    <CodeGroup title="Request" tag="GET" label="/default_secured">
+```python {{ title: 'untyped' }}
+from robyn import Robyn, SubRouter
+
+app = Robyn(__file__)
+
+sub_router = SubRouter("/auth_example")
+
+@sub_router.get("/default_secured")
+def default_secured(request):
+    return "Authentication is set by include_router!"
+
+app.include_router(sub_router, auth_required=True)
+```
+
+```python {{ title: 'typed' }}
+from robyn import Robyn, SubRouter, Request
+
+app = Robyn(__file__)
+
+sub_router = SubRouter("/auth_example")
+
+@sub_router.get("/default_secured")
+def default_secured(request : Request) -> str:
+    return "Authentication is set by include_router!"
+
+app.include_router(sub_router, auth_required=True)
```

Comment:
Hey @IA-PieroCV 👋 

Thank you for the PR 😄 Overall I like the theme of the PR but will have to a review of the code.

However, why do we need an `auth_required` flag in `include_router`? Is it not the same as the SubRouter Class?

Reply:
Yes, actually it is!
This is just to give developers options for including auth_required due to their own criteria.
Of course we can remove any of the `auth_required` instance level parameters, from include_router or the SubRoute object instance.

Reply:
@IA-PieroCV , apologies for the late revert. But, I thinks this is a redundancy. If someone wants to allow auth, they can do it the subrouter class. Could you remove it from here?

Let me know if you have anymore thoughts 😄